# ğŸ§€ mi_cheese
Repository that accompanies the manuscript under the name "Inter-species metabolic interactions in cheese flavour formation"

## ğŸ‘¨ğŸ»â€ğŸ³ Abstract
Cheese fermentation and flavour formation are governed by complex biochemical reactions driven by polymicrobial activity. While the compositional dynamics of cheese microbiomes is relatively well mapped, the mechanistic role of microbial interactions in flavour formation is yet unknown. We microbially and metabolically characterised a year-long Cheddar cheese process using a commonly used starter culture containing Streptococcus thermophilus and Lactococcus strains. By using an experimental strategy whereby certain strains were left out from the starting mixture, we identified the critical role of S. thermophilus in boosting Lactococcus growth and in shaping flavour compound profile. Controlled milk fermentations with systematic exclusion of single Lactococcus strains, combined with genomics, genome-scale metabolic modelling, and metatranscriptomics, indicated that proteolytic activity of S. thermophilus relieves nitrogen limitation for Lactococcus and boosts de novo nucleotide biosynthesis. While S. thermophilus had large contribution to the flavour profile, L. cremoris also played a role by limiting diacetyl and acetoin formation which leads to off-flavour when in excess. This off-flavour control could be attributed to different metabolic re-routing of citrate between L. cremoris and other L. lactis strains. Further, closely related L. lactis strains exhibited different interaction patterns with S. thermophilus highlighting the importance of strain-specificity in cheese-making. Overall, our results bring forward the critical role of competitive and cooperative microbial interactions shaping cheese flavour profile.

## ğŸª¤ Usage & description
Here is availalbe the majority of the processed data, including metabolomics, metatranscriptomics count tables and the corresponding code to reproduce the figures/results of the manuscript. The SLAB genomes (SAMN34041181-SAMN34041202) and the metatranscriptomics reads (SRR24029527-SRR24029544) are available in NCBI under BioProject with PRJNA950467 accession. 

Clone repo:
```
$ git clone https://github.com/Chrats-Melkonian/mi_cheese.git
```
See repo structure:
```
$ tree mi_cheese/ -L 3
mi_cheese/
â”œâ”€â”€ code
â”‚Â Â  â”œâ”€â”€ 01_Figure02.R
â”‚Â Â  â”œâ”€â”€ 02_Figure02.R
â”‚Â Â  â”œâ”€â”€ 03_Figure3panelB_E.R
â”‚Â Â  â”œâ”€â”€ 04_community_GEM_sims.sh
â”‚Â Â  â”œâ”€â”€ 04_figure4panelB.R
â”‚Â Â  â”œâ”€â”€ 04_gf_rxns_compile.sh
â”‚Â Â  â”œâ”€â”€ 04_individual_GEM_sims.py
â”‚Â Â  â”œâ”€â”€ 04_model_summary_compile.sh
â”‚Â Â  â”œâ”€â”€ 04_model_summary_generate.py
â”‚Â Â  â”œâ”€â”€ 05_Figure4panelC.R
â”‚Â Â  â”œâ”€â”€ 06_Figure4panelD_E.R
â”‚Â Â  â”œâ”€â”€ 07_Figure5panelA_C.R
â”‚Â Â  â”œâ”€â”€ 08_Figure05PanelE_supp_plots.R
â”‚Â Â  â”œâ”€â”€ 09_figure5panelF.R
â”‚Â Â  â””â”€â”€ 10_PCAs_supp.R
â”œâ”€â”€ data
â”‚Â Â  â”œâ”€â”€ Models
â”‚Â Â  â”‚Â Â  â”œâ”€â”€ curated
â”‚Â Â  â”‚Â Â  â”œâ”€â”€ data
â”‚Â Â  â”‚Â Â  â”œâ”€â”€ Escher_maps
â”‚Â Â  â”‚Â Â  â””â”€â”€ gapfilled
â”‚Â Â  â”œâ”€â”€ ORF_protein
â”‚Â Â  â”‚Â Â  â”œâ”€â”€ LC.faa
â”‚Â Â  â”‚Â Â  â”œâ”€â”€ LLm1.faa
â”‚Â Â  â”‚Â Â  â”œâ”€â”€ LLm2.faa
â”‚Â Â  â”‚Â Â  â””â”€â”€ ST.faa
â”‚Â Â  â”œâ”€â”€ T0
â”‚Â Â  â”‚Â Â  â”œâ”€â”€ Biochemical
â”‚Â Â  â”‚Â Â  â”œâ”€â”€ metatranscriptomics
â”‚Â Â  â”‚Â Â  â””â”€â”€ table_analysis
â”‚Â Â  â””â”€â”€ T1
â”‚Â Â      â””â”€â”€ Biochemical
â”œâ”€â”€ LICENSE
â””â”€â”€ README.md

```

[![DOI](https://zenodo.org/badge/586860014.svg)](https://zenodo.org/badge/latestdoi/586860014)
